### DISCUSSION TITLE
unexpected column width with X column

### DISCUSSION THREAD

**Discussion by al3atross** [lvjr/tabularray#292] [Q&A]
Hi, 
I would like the first column to be 50% of the total width of the table while the other 50% can be split into multiple columns. I used the X column for this with corresponding ratios.

Why does the width of column A change in this case? Is this a bug or a feature?

```latex
\documentclass{article}
\usepackage{tabularray}

\begin{document}

\begin{tblr}{colspec={X[2]|X[2]}, hlines}
	A & B \\
\end{tblr}

\vspace{5mm}

\begin{tblr}{colspec={X[2]|X[1]|X[1]}, hlines}
	A & B & C\\
\end{tblr}

\end{document}
```
![image](https://user-images.githubusercontent.com/1773071/185582504-52f8184d-57bc-4d64-b6f2-c71e724ea176.png)

**Comment by lvjr** [comment]
`X` columns are not of fixed widths; they just occupy remaining width in proportion to their coefficients. Therefore if you add a letter to one of the cell, the widths of all columns will change. You have to specify widths if you need fixed width columns.

**Comment by al3atross** [comment]
> they just occupy remaining width in proportion to their coefficients

Ok, that is what I like to achieve.
But why does the column width even depend on the `colsep` setting in the following example?

```latex
\documentclass{article}
\usepackage{tabularray}

\begin{document}
	
\begin{tblr}{colspec={X[2]|X[1]|X[1]}, hlines, colsep=6pt}
	A & B & C \\
\end{tblr}

\vspace{5mm}

\begin{tblr}{colspec={X[2]|X[1]|X[1]}, hlines, colsep=12pt}
	A & B & C \\
\end{tblr}

\end{document}
```
![image](https://user-images.githubusercontent.com/1773071/185780387-cee98684-bb4a-4a56-82a3-210765e223f2.png)

How can I achieve the same column width for both `colsep` settings?

What do you mean by:

> Therefore if you add a letter to one of the cell, the widths of all columns will change. 

The content of the cells does not change the width of the columns.

**Comment by lvjr** [comment]
>How can I achieve the same column width for both colsep settings?

You can't. Because `colsep` is not part of the column width, different `colsep` will make different remaining width and make different widths of `X` columns. 

>The content of the cells does not change the width of the columns.

Edited: sorry, it turns out that my memory on `X` columns is inaccurate:  when *all* coefficents of `X` columns are positive, the column widths *are* fixed.

**Comment by lvjr** [comment]
I have created issue https://github.com/lvjr/tabularray/issues/365 for this feature.
